src/group.rs

```diff
@@ -96,6 +108,99 @@ impl Element {
 
         Encoding(bytes)
     }
+
+    /// Create a decaf377 point (X : Y : Z : T) from its Jacobi Quartic representation (s, t)
+    fn from_jacobi_quartic(
+        s: ark_ed_on_bls12_377::Fq,
+        t: ark_ed_on_bls12_377::Fq,
+        sgn: ark_ed_on_bls12_377::Fq,
+    ) -> Element {
```

Comment:
(Later) I think it would make sense to inline this with the other steps, and produce a single set of formulas that compute the Elligator map.  But we could wait to do this until speccing the square root algorithm and the optimized formulas.